### PR TITLE
removes feature gated code for dropping legacy shreds

### PR DESCRIPTION
**Pull request by behzadnouri**
#### Problem
Feature is activated on all clusters.

#### Summary of Changes
Removed feature gated code for dropping legacy shreds.

### CODE REVIEW

File: core/src/shred_fetch_stage.rs [hunk 1]
```diff
@@ -434,170 +425,3 @@ fn check_feature_activation(
...
-        super::*,
-        solana_ledger::{
-            blockstore::MAX_DATA_SHREDS_PER_SLOT,
-            shred::{ReedSolomonCache, Shred, ShredFlags},
-        },
-        solana_sdk::packet::Packet,
-    };
-
-    #[test]
-    fn test_data_code_same_index() {
```
Comment (behzadnouri) [inline core/src/shred_fetch_stage.rs:450]: This test is outdated.
It used to verify that the deduper does not discard shreds with the same index but different types; but the deduper is moved to sigverify.
The test is creating legacy shreds which are now always discarded, so I removed the test.

File: core/src/shred_fetch_stage.rs [hunk 2]
```diff
@@ -434,170 +425,3 @@ fn check_feature_activation(
...
-            max_slot,
-            shred_version,
-            |_| false, // should_drop_legacy_shreds
-            |_| true,  // enable_chained_merkle_shreds
-            &mut stats,
-        ));
-    }
-
-    #[test]
-    fn test_shred_filter() {
```
Comment (behzadnouri) [inline core/src/shred_fetch_stage.rs:500]: This test is a duplicate of `ledger::shred::test_should_discard_shred`.
The test is creating legacy shreds which are now always discarded, so I removed the test.

**Review by AshwinSekar — APPROVED** [review]

**Review by steviez — COMMENTED** [review]
Was a little late but looks good to me.

Just to confirm - the check for legacy happens at fetch stage, so we are fine to continue to have a bunch of unit tests (ie `Blockstore`) that insert legacy shreds right ?

**Comment by behzadnouri** [comment]
> Just to confirm - the check for legacy happens at fetch stage, so we are fine to continue to have a bunch of unit tests (ie `Blockstore`) that insert legacy shreds right ?

yeah, but we probably should migrate them off legacy shreds at some point anyways.

**Pull request merged**